symphony/bdk/core/service/message/message_service.py

```diff
@@ -419,3 +421,58 @@ async def get_message_relationships(
             "user_agent": ""
         }
         return await self._default_api.v1_admin_messages_message_id_metadata_relationships_get(**params)
+
+    @retry
+    async def search_messages(self, query: MessageSearchQuery, sort_dir: str = "desc", skip: int = 0,
+                              limit: int = 50) -> List[V4Message]:
+        """Searches for messages in the context of a specified user, given an argument-based query and pagination
+        attributes (skip and limit parameters).
+        See: `Message Search (using POST) <https://developers.symphony.com/restapi/reference#message-search-post>`_
+
+        :param query: The search query arguments
+        :param sort_dir: Sorting direction for response. Possible values are desc (default) and asc.
+        :param skip: Number of messages to skip. Default: 0
+        :param limit: Maximum number of messages to return. Default: 50
+        :return: The list of matching messages in the stream
+        """
+        MessageService._validate_message_search_query(query)
+        params = {
+            "session_token": await self._auth_session.session_token,
+            "key_manager_token": await self._auth_session.key_manager_token,
+            "query": query,
+            "sort_dir": sort_dir,
+            "skip": skip,
+            "limit": limit
+        }
+        message_list = await self._messages_api.v1_message_search_post(**params)
+        return message_list.value  # endpoint returns empty list when no values found
+
+    async def search_all_messages(self, query: MessageSearchQuery, sort_dir: str = "desc", chunk_size: int = 50,
```

Comment:
should this one be annotated with @retry too? or because we reuse the other it is covered?